src/EventSubscriber/RequestChannelSubscriber.php

```
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\ShopApiPlugin\EventSubscriber;
+
+use Sylius\ShopApiPlugin\Checker\ChannelExistenceCheckerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+final class RequestChannelSubscriber implements EventSubscriberInterface
+{
+    /** @var ChannelExistenceCheckerInterface */
+    private $channelExistenceChecker;
+
+    public function __construct(ChannelExistenceCheckerInterface $channelExistenceChecker)
+    {
+        $this->channelExistenceChecker = $channelExistenceChecker;
+    }
+
+    public function checkChannelCode(FilterControllerEvent $event): void
+    {
+        $requestAttributes = $event->getRequest()->attributes;
+
+        if (!$requestAttributes->has('channelCode')) {
+            return;
```

Is it really true? Maybe we should throw an exception if no channel code found as well

---

Unfortunately, there still could be some routes that do not have a channel code in the URL (for example for logging in). I can of course change that, but it would be a big commitment from us :) Maybe we should take it into the consideration and implement such a feature in separate PR?